ofctrl/transaction.go

```diff
@@ -121,23 +121,29 @@ func (tx *Transaction) listenReply() {
 	for {
 		select {
 		case reply, ok := <-tx.controlReplyCh:
-			if !ok { // controlReplyCh closed.
-				return
-			}
-			switch reply.msgType {
-			case BundleControlMessage:
-				select {
-				case tx.controlIntCh <- reply:
-				case <-time.After(messageTimeout):
-					log.Warningln("BundleControlMessage reply message accept timeout")
+			if func() bool {
+				tx.lock.Lock()
+				defer tx.lock.Unlock()
+
+				if !ok { // controlReplyCh closed.
+					return true
```

Comment:
Does it need to hold the lock?

Reply:
Just found we don't need to lock that many, updated the lock scope.